openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java

```diff
@@ -129,26 +132,70 @@ public static String toGPX(RoutingRequest rreq, RouteResult[] routeResults) thro
+                orsMail.setDomain("@" + mail[1]);
+                orsMail.setId(mail[0]);
+                orsPerson.setEmail(orsMail);
+            } catch (Exception e) {
+                orsMail.setDomain("");
+                orsMail.setId("");
+                orsPerson.setEmail(orsMail);
+            }
+        } else {
+            orsMail.setDomain("");
+            orsMail.setId("");
+            orsPerson.setEmail(orsMail);
+            new MissingConfigParameterException("support_mail");
+        }
+
         LinkType orsLink = new LinkType();
-        orsLink.setHref(AppConfig.Global().getParameter("info", "base_url"));
-        orsLink.setText(AppConfig.Global().getParameter("info", "base_url"));
-        orsLink.setType("text/html");
-        orsPerson.setLink(orsLink);
-        orsPerson.setName(AppConfig.Global().getParameter("info", "author_tag"));
+        // set base_url
+        if (AppConfig.Global().getParameter("info", "base_url") != null) {
+            orsLink.setHref(AppConfig.Global().getParameter("info", "base_url"));
+            orsLink.setText(AppConfig.Global().getParameter("info", "base_url"));
+            orsLink.setType("text/html");
+            orsPerson.setLink(orsLink);
+        } else {
+            orsLink.setHref("");
+            orsLink.setText("");
+            orsLink.setType("text/html");
+            orsPerson.setLink(orsLink);
+            new MissingConfigParameterException("base_url");
+        }
+
+        // set author_tag
+        if (AppConfig.Global().getParameter("info", "author_tag") != null) {
+            orsPerson.setName(AppConfig.Global().getParameter("info", "author_tag"));
+        } else {
+            orsPerson.setName("");
+            new MissingConfigParameterException("author_tag");
+        }
         metadata.setAuthor(orsPerson);
-        // create and set copyright
+
+        // set copyright
         CopyrightType copyright = new CopyrightType();
-        copyright.setAuthor(RoutingServiceSettings.getAttribution());
-        copyright.setLicense(AppConfig.Global().getParameter("info", "content_licence"));
+        if (RoutingServiceSettings.getAttribution() != null) {
+            copyright.setAuthor(RoutingServiceSettings.getAttribution());
+
+        } else {
+            copyright.setAuthor("");
+            new MissingConfigParameterException("attribution");
+        }
+        // set content_licence
+        if (AppConfig.Global().getParameter("info", "content_licence") != null) {
+            copyright.setLicense(AppConfig.Global().getParameter("info", "content_licence"));
+        } else {
+            copyright.setLicense("");
+            new MissingConfigParameterException("content_licence");
+        }
         // create and set current date as XMLGregorianCalendar element
         Date date = new Date();
         GregorianCalendar c = new GregorianCalendar();
```

Comment:
Can we not use general java date here rather than a new object? `int year = Calendar.getInstance().get(Calendar.YEAR);` should work as far as I know

Reply:
Ignore this for now - we can update later if needed